artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/CoreAmqpConverter.java

```diff
@@ -425,7 +419,7 @@ private static Section convertBody(ServerJMSMessage message, Map<Symbol, Object>
                break;
          }
       } else if (message instanceof ServerJMSTextMessage) {
-         String text = (((TextMessage) message).getText());
+         String text = (((ServerJMSTextMessage) message).getText());
```

Comment:
Can we use an interface here. And not concrete classes

Reply:
No.. we can't... we need to remove the JMS API implementation from the converter.

Right now the converter is depending on the JMS Client API.

Wildfly for instance is not JMS any longer. It is just simpler to remove the dependency.

This is again just for conversion.

Reply:
I got that you want to remove the current api. But still can have an interface even if an internal one

Reply:
I would then duplicate the whole JMS API internally for the conversion. 

Concrete types is all we need here. The only reason I added these classes in the first place was to do the conversion.

Reply:
So either then move away from it being JMS this and that. E.g. to convert from/to Core which core is not JMS. Lets just drop JMS properly

Reply:
Or even why not just move over to being the Jakarta EE interfaces that replace the older oracle / geronimo ones. As i understand its 1..1 atm

Reply:
the AMQP Protocol Head should not depend on JMS at all.. no need for it just for the interfaces definition that are not required.

Reply:
Then lets drop the JMS constructs here then not this halfway house